### PR TITLE
Add stage-3 plugin option

### PR DESCRIPTION
**Pull request by existentialism**
Adds support for enabling transformation of **shipping** stage 3 features.

**EDIT** Updating this to:

* Enable syntax-only if transform is not needed
* Update mapping for optional catch binding (https://github.com/kangax/compat-table/pull/1159)

### CODE REVIEW

**Comment by chicoxyzzy** [comment]
> 🚲 🏠 the option name :)

I think `unspeccedProposals` may be a good name for it

File: data/plugins.json [hunk 1]
```diff
@@ -238,5 +237,14 @@
     "ios": "10",
     "opera": "45",
     "electron": "1.7"
-  }
+  },
+  "transform-async-generator-functions": {},
```
Comment (existentialism) [inline data/plugins.json]: These will get filled out once we land https://github.com/babel/babel-preset-env/pull/402

File: src/index.js [hunk 2]
```diff
@@ -7,13 +7,36 @@ import { defaultWebIncludes } from "./default-includes";
...
-import { prettifyTargets, semverify } from "./utils";
+import { filterStageFromList, prettifyTargets, semverify } from "./utils";
 import type { Plugin, Targets } from "./types";
 
+const getPlugin = (pluginName: string) => {
+  const plugin = availablePlugins[pluginName];
+
+  if (!plugin) {
+    throw new Error(
+      `Could not find plugin "${pluginName}". Ensure there is an entry in ./available-plugins.js for it.`,
```
Comment (existentialism) [inline src/index.js]: This kept tripping me up 😜

Comment (yavorsky) [reply src/index.js]: lol

**Comment by existentialism** [comment]
I'll create a followup to update README, but we should def discuss if there's a better name for the option too.

**Pull request merged**